File: Domain/Models/Photo.cs
using System.Text.Json.Serialization;

namespace Domain.Models
{
    public class Photo
    {
        public string Id { get; set; }
        public string Url { get; set; }
        public bool IsMain { get; set; }
        public int CategoryId { get; set; } 
        public Category Category { get; set; } 
        public string PhotoDescription { get; set; }
        public string PhotoTitle { get; set; }
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
        public string UserId { get; set; }
        public DateTime UploadDate { get; set; }
    }
}

File: client-app/src/app/layout/models/photo.ts
import { User } from "./user";

export interface Photo {
  id: string;
  url: string;
  isMain: boolean;
  photoDescription: string;
  photoTitle: string;
  categoryId: number;
  username: string;
  uploadDate: Date;
}

export class Photo implements Photo {
  constructor(user: User) {
    this.username = user.userName;
  }
}
